File: 2244-number-of-laser-beams-in-a-bank/number-of-laser-beams-in-a-bank.dart
class Solution {
  int numberOfBeams(List<String> bank) {
    int res = 0;
    int preLazer = 0;
    for(String row in bank){
      int curLazer = 0;
      for(String i in row.split('')) if(i=='1') curLazer++;
      res += curLazer * preLazer;
      if(curLazer!=0) preLazer = curLazer;
    }
    return res;
  }
}

File: 2244-number-of-laser-beams-in-a-bank/number-of-laser-beams-in-a-bank.rs
impl Solution {
    pub fn number_of_beams(mut bank: Vec<String>) -> i32 {
        let (mut res,mut preLazer)= (0,0);
        bank.reverse();
        for row in bank {
            let mut curLazer = 0;
            for i in row.chars() {
                if i=='1' {
                    curLazer +=1;
                }
            }
            res += curLazer*preLazer;
            if curLazer != 0 {
                preLazer = curLazer;
            }
        }
        res
    }
}